app/processing/settlement/get-last-settlement.js

```diff
@@ -0,0 +1,31 @@
+const db = require('../../data')
+
+const getLastSettlement = async (settlement, transaction) => {
+  return db.settlement.findOne({
+    transaction,
+    attributes: [
+      'paymentRequestId',
+      'invoiceNumber',
+      'reference',
+      'settled',
+      'settlementDate',
+      'value'
+    ],
+    where: {
+      settlementDate: {
+        [db.Sequelize.Op.lt]: settlement.settlementDate
+      },
+      value: {
+        [db.Sequelize.Op.lt]: settlement.value
```

Comment:
Value must be lower.  If we didn't have this then in a BACS Rejection scenario we'd get a redundant version of the current settlement with an earlier date.  The impact of this would be we'd incorrectly calculate the current payment as 0.